MoreLinq/TakeLast.cs

```diff
@@ -52,23 +53,9 @@ public static IEnumerable<TSource> TakeLast<TSource>(this IEnumerable<TSource> s
             return
                 source.TryGetCollectionCount() is int collectionCount
                 ? source.Slice(Math.Max(0, collectionCount - count), int.MaxValue)
-                : _(); IEnumerable<TSource> _()
-                {
-                    if (count <= 0)
-                        yield break;
-
-                    var q = new Queue<TSource>(count);
-
-                    foreach (var item in source)
-                    {
-                        if (q.Count == count)
-                            q.Dequeue();
-                        q.Enqueue(item);
-                    }
-
-                    foreach (var item in q)
-                        yield return item;
-                }
+                : source.CountDown(count, (x, cd) => (Element: x, CountDown: cd ))
```

Comment:
The element name for the second tuple item should be `Countdown` instead of `CountDown`. In English, _countdown_ is a noun for the counter but _count-down_ is the verb for or action of counting-down. You use Pascal-casing of `CountDown` due to the hyphenation; devil in the details. :wink: